### PR TITLE
Update module: bakta/bakta

### PR DESCRIPTION
**Pull request by jasmezz**
- Update bakta 1.8.2 -> 1.9.2
- Add nf-test
- Remove old pytest files from bakta/baktadbdownload

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bakta/bakta/main.nf
Comment (jfy133) [inline modules/nf-core/bakta/bakta/main.nf:38]: Is this required for BATKA?

File: modules/nf-core/bakta/bakta/tests/main.nf.test
Comment (jfy133) [inline modules/nf-core/bakta/bakta/tests/main.nf.test:52]: Really? Nothing is consistent, nor has any stable strings that can be used to confirm the file isn't just empty?

**Review by jfy133 — COMMENTED** [review]
Almost there but  I am confused by the AMRfinderplus command, and also I'm concerned the tests are too lax...

File: modules/nf-core/bakta/bakta/main.nf
Comment (jasmezz) [reply modules/nf-core/bakta/bakta/main.nf:38]: Tests now pass without this line. I had to add it because some DB files are not in the downloaded folder but are created by `amrfinder_update` (similar to the antismash vs. antismash-lite issue). I don't know what changed but I'll remove the line. If I ever encounter DB issues with bakta I hopefully remember this comment/commit ;)

File: modules/nf-core/bakta/bakta/tests/main.nf.test
Comment (jasmezz) [reply modules/nf-core/bakta/bakta/tests/main.nf.test:52]: Right, the pytest didn't check for content as well, thanks for questioning that. There are timestamps, but also other reliable strings! Will update.

File: modules/nf-core/bakta/bakta/main.nf
Comment (jfy133) [reply modules/nf-core/bakta/bakta/main.nf:38]: :tada:

File: modules/nf-core/bakta/bakta/tests/main.nf.test
Comment (jfy133) [reply modules/nf-core/bakta/bakta/tests/main.nf.test:52]: Huh ok, well strings best if possible :)

**Review by jfy133 — APPROVED** [review]
LGTM now if tests are passing

**Comment by jasmezz** [comment]
Okay seems like we have to use the stub run only. This was also the case with the pytest; the conda CI checks wouldn't be happy even with the `process_high` label. I will revert back to `process_medium`.
But I can at least confirm that the conda run of `nf-core modules test bakta/baktadbdownload` was successful on my local machine. :heavy_check_mark:

**Comment by jasmezz** [comment]
Waiting for you go/comment on stub-only before merge :)

**Comment by jfy133** [comment]
@jasmezz you need to turn on the actual test too ;)

**Comment by jasmezz** [comment]
What is the maximum we can go with memory requirement for CI tests? It's not happy with even 14 GB. It works on my local machine (64 GB, I hope this is not necessary...)

**Comment by jfy133** [comment]
Almost there! We have a way to ignore conda tests, if we can't push it over the edge?

**Comment by jasmezz** [comment]
15.6 GB is maximum, GH won't let me go higher. So I would either just skip conda CI checks or stub all tests (as with pytest). I searched the docs and Slack help channel but cannot find a syntax to skip conda. Can you lead me in the right direction?

**Comment by jfy133** [comment]
> 15.6 GB is maximum, GH won't let me go higher. So I would either just skip conda CI checks or stub all tests (as with pytest). I searched the docs and Slack help channel but cannot find a syntax to skip conda. Can you lead me in the right direction?

OK! I'm a bit confused why conda is failing here (maybe something to do with how DIAMOND works in containers), but that we can have real tests is preferred!

If I remember correctly, there is a long list at the bottom of the `ci.yml` where you can specify to skip conda on this module

**Comment by jasmezz** [comment]
Hm there is no `ci.yml` in modules (only on pipeline level). The only file I found related is `.github/workflows/test.yml`. I set bakta/bakta to use profile singularity (see latest commit). However, conda CI test is still there, but singularity is now missing :see_no_evil:

**Comment by jfy133** [comment]
Ah `test.yml` could be the one! It shoudl be an exclude list not a run list?

**Comment by jfy133** [comment]
I can have alook

File: .github/workflows/test.yml
Comment (jfy133) [inline .github/workflows/test.yml:479]:
```suggestion
          - profile: conda
```

**Comment by jasmezz** [comment]
Ahh sorry, I didn't expect you to interfere with my branch :laughing: Now double commit. Matthias told me about the exclude/notinclude list at the same time as you.

**Comment by jasmezz** [comment]
Any further comments before merge @jfy133? :)

**Comment by jfy133** [comment]
Nope, good to go!

File: modules/nf-core/bakta/bakta/main.nf
Comment (cahuparo) [reply modules/nf-core/bakta/bakta/main.nf]: Hi @jfy133 and @jasmezz,

I am running into this issue with the `amrfinder_update`. The `BAKTA_BAKTADBDOWNLOAD` command fails to download the amrfinderplus-db. So when `bakta` tries to find this database, it fails:

```nextflow
  Traceback (most recent call last):
    File "/local/cluster/micromamba/envs/bakta-1.9.1/bin/bakta", line 10, in <module>
      sys.exit(main())
    File "/local/cluster/micromamba/envs/bakta-1.9.1/lib/python3.8/site-packages/bakta/main.py", line 275, in main
      expert_amr_found = exp_amr.search(cdss, cds_aa_path)
    File "/local/cluster/micromamba/envs/bakta-1.9.1/lib/python3.8/site-packages/bakta/expert/amrfinder.py", line 47, in search
      raise Exception(f"amrfinder error! error code: {proc.returncode}. Please, try 'amrfinder_update --force_update --database {amrfinderplus_db_path}' to update AMRFinderPlus's internal database.")
  Exception: amrfinder error! error code: 1. Please, try 'amrfinder_update --force_update --database /nfs5/BPP/Grunwald_Lab/home/paradarc/pathogensurveillance/work/3d/19178e85f70539076ec38ddcf5bd0f/db/amrfinderplus-db' to update AMRFinderPlus's internal database.
``` 
Do you recommend to add this line to the backta's `main.nf`? or do you have any idea on how to get it to work?

` amrfinder_update --force_update --database $db/amrfinderplus-db`

For context, I working on the [pathogensurveillance](https://github.com/nf-core/pathogensurveillance) pipeline.

Thanks!

Camilo

Comment (jasmezz) [reply modules/nf-core/bakta/bakta/main.nf]: Hi @cahuparo, did you use bakta (and baktadbdownload) with default parameters or change anything? I am trying to replicate your error, currently unsuccessfully (using either conda or singularity for the test runs). I would like to have a reproducible case before adding the `amrfinder_update` line back in, because that would slow the module down.

Comment (cahuparo) [reply modules/nf-core/bakta/bakta/main.nf]: Hi @jasmezz, 

when I ran the pipeline from a brand new directory I saw the error I showed above. For that, my command was:

```bash
nextflow run main.nf --input https://raw.githubusercontent.com/grunwaldlab/pathogensurveillance/master/test/data/metadata_small.csv --outdir test_out --download_bakta_db true -profile mamba -resume
``` 
So I added the line for the `amrfinder_update` to the bakta `main.nf` this way:

```nextflow
    """
    /nfs5/BPP/Grunwald_Lab/home/paradarc/pathogensurveillance/work/conda/bakta_baktadbdownload-cfec0aac1e1132b000975a91ee5beb7d/bin/amrfinder_update --force_update --database $db/amrfinderplus-db

    bakta \\
        $fasta \\
        $args \\
        --threads $task.cpus \\
        --prefix $prefix \\
        $proteins_opt \\
        $prodigal_tf \\
        --db $db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bakta: \$(echo \$(bakta --version) 2>&1 | cut -f '2' -d ' ')
    END_VERSIONS
    """
``` 
Adding the line actually worked. So we are debating if we need to create a local `main.nf` to handle that. But seems to only happen with the conda/mamba installation for some reason.

Thanks, 

Camilo

Comment (jasmezz) [reply modules/nf-core/bakta/bakta/main.nf]: Hi @cahuparo, I finally found time for this! I fixed it in #5048.

**Pull request merged**
